### PR TITLE
[IMP] core: new API to check access rights and rules

### PR DESCRIPTION
**Pull request by rco-odoo**
Companion of https://github.com/odoo/odoo/pull/179148.

### CODE REVIEW

**Review by kmagusiak — APPROVED** [review]

File: content/developer/tutorials/restrict_data_access.rst [hunk 1]
```diff
@@ -355,21 +355,15 @@ Explicit security checks can be performed by:
...
-  verifies that the current user is allowed to perform the operation on *every*
-  record of the set.
-
-.. warning:: Checking access rights and checking record rules are separate
-             operations, if you're checking record rules you usually want to
-             also check access rights beforehand.
+* Calling ``check_access(operations)`` on a recordset, this verifies that the
+  current user is allowed to perform the operation on *every* record of the set.
+  As a special case, when the recordset is empty, it verifies that the current
+  user has some permission to perform the operation on the model in general.
```
Comment (AntoineVDV) [inline content/developer/tutorials/restrict_data_access.rst:361]: I'm not sure to understand what you mean by "some permission"? As in "access through some (at least one) ACL"?

Comment (rco-odoo) [reply content/developer/tutorials/restrict_data_access.rst:361]: That's it.  Some permission means some ACL.

File: content/developer/tutorials/restrict_data_access.rst [hunk 2]
```diff
@@ -355,21 +355,15 @@ Explicit security checks can be performed by:
...
-  verifies that the current user is allowed to perform the operation on *every*
-  record of the set.
-
-.. warning:: Checking access rights and checking record rules are separate
-             operations, if you're checking record rules you usually want to
-             also check access rights beforehand.
+* Calling ``check_access(operations)`` on a recordset, this verifies that the
+  current user is allowed to perform the operation on *every* record of the set.
+  As a special case, when the recordset is empty, it verifies that the current
+  user has some access right to perform the operation on the model in general.
```
Comment (AntoineVDV) [inline content/developer/tutorials/restrict_data_access.rst:361]:
```suggestion
  user has some access rights to perform the operation on the model in general.
```

**Review by AntoineVDV — APPROVED** [review]
@robodoo delegate+

**Comment by rco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**